### PR TITLE
Create function to read keys - Closes #5233

### LINKED ISSUE #5233: Create function to read keys
### Description

Create `readKey(bytes, schema) => object`

### Motivation

Messages are encoded as key-value pairs we need a function to read keys from a binary message based on the [LIP](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0027.md#keys).

### Acceptance Criteria

- `readKey()` is able to read keys to provide information on how to read a binary message as per the LIP
- Code should be optimised for speed as much as possible
- Full tests

### Additional Information

Related issues: #5200

### PR DESCRIPTION
**Pull request by ManuGowda**
### What was the problem?

This PR resolves #5233 

### How was it solved?

- Implemented `readKey` function to get `fieldNumber, wireType`
### How was it tested?

- Added unit test
- Added benchmark test for performace

### CODE REVIEW

File: elements/lisk-codec/benchmark/buffer_concat.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+const { Suite } = require('benchmark');
+const { randomBytes } = require('crypto');
+const suite = new Suite();
+const buf1 = randomBytes(1024);
+const buf2 = randomBytes(1024);
+
+suite
+  .add('Buffer.concat', () => {
+    Buffer.concat([buf1, buf2]);
```
Comment (shuse2) [inline elements/lisk-codec/benchmark/buffer_concat.js:24]: I think this is not needed? if we gonna keep this, probably nice to test with more elements and also with totalLength and without totalLength for the second parameter on the `Buffer.concat`

Comment (ManuGowda) [reply elements/lisk-codec/benchmark/buffer_concat.js:24]: ok should i remove this now?

Comment (shuse2) [reply elements/lisk-codec/benchmark/buffer_concat.js:24]: Either way it's fine

**Comment by ManuGowda** [comment]
> `keys.ts` and `readkey.spec.ts` look good to me.
> I have to admit, I didn't really understand some of the other files (like `buffer_concat.js `)

the `buffer_concat` came out of discussion we were having about performance impacts, its not relevant to this PR, so i removed it from this PR and added to #5303

**Pull request merged**